### DISCUSSION TITLE
How can i set status for response when fetching data?

### DISCUSSION THREAD

**Discussion by oh-klahoma** [vercel/next.js#50004] [App Router]
Pages routing:
```js
export const getServerSideProps = async ({ res }) => {
  const { error } = await (....)
  
  if(error) {
     res.statusCode = 404;
  }

  return {
    props: {},
  };
};
```

How can i do this with **app routing**?

**Comment by devjiwonchoi** [comment]
For the code you've provided, you could do this by using [notFound()](https://nextjs.org/docs/app/api-reference/functions/not-found).
Also, for other status codes such as errors, you can manage it by [Error handling](https://nextjs.org/docs/app/building-your-application/routing/error-handling).

In addition, you could set `not-found.tsx` to the /app directory and this doesn't require to call notFound() in pages.

**Reply by oh-klahoma** [reply]
I want setting status for the response. Now I see that `not-found.tsx` also returns status 200 instead of  404

**Reply by oh-klahoma** [reply]
Now i checking this in production build and this work fine. For `not-found` - 404 status code
But I would like to know if it is possible to change the status for response

**Reply by devjiwonchoi** [reply]
What's your intention for that?
